### ISSUE TITLE
rust-1.74.0-beta fails to execute x.py install with: thread 'main' panicked at install.rs:59:17 - Failed the write access check for the current user. No such file or directory

### ISSUE THREAD

**Issue by stefson** [rust-lang/rust#117203]
hey everyone, I get this message from `x.py` when `install` is executed: 

```

       Fresh bootstrap v0.0.0 (/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/src/bootstrap)
    Finished dev [unoptimized] target(s) in 0.50s
running: /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/bootstrap/debug/bootstrap install -vv --config=/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/config.toml -j12
warning: x.py has made several changes recently you may want to look at
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, add `changelog-seen = 2` at the top of `config.toml`
finding compilers
CC_x86_64-gentoo-linux-musl = "x86_64-gentoo-linux-musl-gcc"
CFLAGS_x86_64-gentoo-linux-musl = ["-ffunction-sections", "-fdata-sections", "-fPIC", "-m64", "-pipe"]
CXX_x86_64-gentoo-linux-musl = "x86_64-gentoo-linux-musl-g++"
CXXFLAGS_x86_64-gentoo-linux-musl = ["-ffunction-sections", "-fdata-sections", "-fPIC", "-m64", "-pipe"]
AR_x86_64-gentoo-linux-musl = "x86_64-gentoo-linux-musl-ar"
running sanity check
learning about cargo
> Assemble { target_compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl } }
  > Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    > Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
    < Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
    > Rustc { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [] }
      > Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
        > StartupObjects { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        < StartupObjects { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
        > Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
          > Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage0-sysroot to avoid caching bugs
          < Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage0-sysroot/lib/rustlib/x86_64-gentoo-linux-musl/lib to avoid caching bugs
          c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
        < Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
        c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
        > StdLink { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
          c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
          c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        < StdLink { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
      < Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
      c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
      c Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      > Llvm { target: x86_64-gentoo-linux-musl }
      < Llvm { target: x86_64-gentoo-linux-musl }
      c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
      > RustcLink { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [] }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < RustcLink { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [] }
    < Rustc { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [] }
    > Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage1 to avoid caching bugs
    < Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    > Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage1/lib/rustlib/x86_64-gentoo-linux-musl/lib to avoid caching bugs
    < Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Llvm { target: x86_64-gentoo-linux-musl }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
  < Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
  > Rustc { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [] }
    > Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      > StartupObjects { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < StartupObjects { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
      c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
      > StdLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
        c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < StdLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
    < Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
    c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    c Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
    c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Llvm { target: x86_64-gentoo-linux-musl }
    c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    > RustcLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [] }
      c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    < RustcLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [] }
  < Rustc { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [] }
  > Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage2 to avoid caching bugs
  < Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
  c Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
  c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
  > Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage2/lib/rustlib/x86_64-gentoo-linux-musl/lib to avoid caching bugs
  < Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
  c Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
  c Llvm { target: x86_64-gentoo-linux-musl }
  c Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
< Assemble { target_compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl } }
> Std { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
  > Std { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    > Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      > StartupObjects { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < StartupObjects { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
      c Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      c Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      > StdLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
        c Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < StdLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
    < Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
    c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
    > RustInstaller { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      > ToolBuild { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, tool: "rust-installer", path: "src/tools/rust-installer", mode: ToolBootstrap, is_optional_tool: false, source_type: InTree, extra_features: [], allow_features: "" }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
        c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      < ToolBuild { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, tool: "rust-installer", path: "src/tools/rust-installer", mode: ToolBootstrap, is_optional_tool: false, source_type: InTree, extra_features: [], allow_features: "" }
    < RustInstaller { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
  < Std { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
thread 'main' panicked at install.rs:59:17:
Failed the write access check for the current user. No such file or directory (os error 2)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/./x.py", line 50, in <module>
    bootstrap.main()
  File "/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 1121, in main
    bootstrap(args)
  File "/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 1097, in bootstrap
    run(args, env=env, verbose=build.verbose, is_bootstrap=True)
  File "/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 187, in run
    raise RuntimeError(err)
RuntimeError: failed to run: /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/bootstrap/debug/bootstrap install -vv --config=/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/config.toml -j12
```

It seems there are some backports still missing for beta, since I can fix this by reverting https://github.com/rust-lang/rust/pull/116127 locally, and there are some subsequent fixes in nightly who seemed to have fixed the issue for good. 

So, is there somewhere a list of the nightly backports needed to fix this problem? Thanks.

**Comment by stefson** [comment]
@nikic can I ping you on this? Beta is not compiling on gentoo, I assume at least one patch from nightly still needs to be backported to beta.

**Comment by onur-ozkan** [comment]
What is HEAD? https://github.com/rust-lang/rust/pull/116349 should have fixed that.

**Comment by stefson** [comment]
Hi, I'm on latest beta from last week. If I apply #116349 locally and compile, I get access denied error: 

```
running: /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/bootstrap/debug/bootstrap install -vv --config=/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/config.toml -j12
warning: x.py has made several changes recently you may want to look at
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, add `changelog-seen = 2` at the top of `config.toml`
finding compilers
CC_x86_64-gentoo-linux-musl = "x86_64-gentoo-linux-musl-gcc"
CFLAGS_x86_64-gentoo-linux-musl = ["-ffunction-sections", "-fdata-sections", "-fPIC", "-m64", "-pipe"]
CXX_x86_64-gentoo-linux-musl = "x86_64-gentoo-linux-musl-g++"
CXXFLAGS_x86_64-gentoo-linux-musl = ["-ffunction-sections", "-fdata-sections", "-fPIC", "-m64", "-pipe"]
AR_x86_64-gentoo-linux-musl = "x86_64-gentoo-linux-musl-ar"
running sanity check
learning about cargo
> Assemble { target_compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl } }
  > Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    > Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
    < Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
    > Rustc { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [] }
      > Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
        > StartupObjects { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        < StartupObjects { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
        > Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
          > Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage0-sysroot to avoid caching bugs
          < Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage0-sysroot/lib/rustlib/x86_64-gentoo-linux-musl/lib to avoid caching bugs
          c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
        < Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
        c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
        > StdLink { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
          c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
          c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        < StdLink { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
      < Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
      c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
      c Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      > Llvm { target: x86_64-gentoo-linux-musl }
      < Llvm { target: x86_64-gentoo-linux-musl }
      c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
      > RustcLink { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [] }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < RustcLink { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [] }
    < Rustc { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, crates: [] }
    > Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage1 to avoid caching bugs
    < Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    > Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage1/lib/rustlib/x86_64-gentoo-linux-musl/lib to avoid caching bugs
    < Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Llvm { target: x86_64-gentoo-linux-musl }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
  < Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
  > Rustc { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [] }
    > Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      > StartupObjects { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < StartupObjects { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
      c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
      > StdLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
        c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < StdLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
    < Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
    c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    c Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
    c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Sysroot { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, force_recompile: false }
    c Llvm { target: x86_64-gentoo-linux-musl }
    c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    > RustcLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [] }
      c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    < RustcLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [] }
  < Rustc { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [] }
  > Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage2 to avoid caching bugs
  < Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
  c Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
  c Libdir { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
  > Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
Removing sysroot /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/x86_64-gentoo-linux-musl/stage2/lib/rustlib/x86_64-gentoo-linux-musl/lib to avoid caching bugs
  < Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
  c Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
  c Llvm { target: x86_64-gentoo-linux-musl }
  c Sysroot { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, force_recompile: false }
< Assemble { target_compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl } }
> Std { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
  > Std { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    > Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      > StartupObjects { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < StartupObjects { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
      c Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
      c Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      > StdLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
        c Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Libdir { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      < StdLink { compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl }, target_compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, crates: [], force_recompile: false }
    < Std { target: x86_64-gentoo-linux-musl, compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, crates: [], force_recompile: false }
    c Assemble { target_compiler: Compiler { stage: 1, host: x86_64-gentoo-linux-musl } }
    c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
    > RustInstaller { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
      > ToolBuild { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, tool: "rust-installer", path: "src/tools/rust-installer", mode: ToolBootstrap, is_optional_tool: false, source_type: InTree, extra_features: [], allow_features: "" }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
        c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
        c Sysroot { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, force_recompile: false }
      < ToolBuild { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl, tool: "rust-installer", path: "src/tools/rust-installer", mode: ToolBootstrap, is_optional_tool: false, source_type: InTree, extra_features: [], allow_features: "" }
    < RustInstaller { compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
    c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-gentoo-linux-musl } }
  < Std { compiler: Compiler { stage: 2, host: x86_64-gentoo-linux-musl }, target: x86_64-gentoo-linux-musl }
 * ACCESS DENIED:  mkdir:              /usr/lib/rust/1.74.0_beta
thread 'main' panicked at install.rs:78:5:
User doesn't have write access on `install.prefix` path in the `config.toml`.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/./x.py", line 50, in <module>
    bootstrap.main()
  File "/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 1121, in main
    bootstrap(args)
  File "/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 1097, in bootstrap
    run(args, env=env, verbose=build.verbose, is_bootstrap=True)
  File "/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 187, in run
    raise RuntimeError(err)
RuntimeError: failed to run: /var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/build/bootstrap/debug/bootstrap install -vv --config=/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc-beta-src/config.toml -j12

```

**Comment by onur-ozkan** [comment]
```txt
 * ACCESS DENIED:  mkdir:              /usr/lib/rust/1.74.0_beta
thread 'main' panicked at install.rs:78:5:
User doesn't have write access on `install.prefix` path in the `config.toml`.
```

Logs says you don't have permission to put files into `install.prefix` path from your build configuration. Without #116349 you would still get `permission denied` error.

**Comment by stefson** [comment]
I believe this is a sandbox error, so x.py install tries to write to / instead of it's prefix, and this is caused either by #116349 or it was already introduced by the older #116127 

If I revert them both locally, I get a working x.py install

**Comment by onur-ozkan** [comment]
Interesting :thinking: Can you share your build configuration?

**Comment by stefson** [comment]
it basically does this: 
```
DESTDIR="${D}" "${EPYTHON}" ./x.py install -vv --config="${S}"/config.toml -j$(makeopts_jobs) || die
```
which translates to: 
DESTDIR="/var/tmp/portage/dev-lang/rust-1.74.0_beta/image/" python3.11 ./x.py install -vv --config="/var/tmp/portage/dev-lang/rust-1.74.0_beta/work/rustc_beta-src/config.toml" -j12 || die

and the patch seems to make x.py ignore the DESTDIR="" command

In the rust context, the build config you asked me to share is the config.toml? Have it attached for you, please feel free to ask for any additional information.
[config.toml.gz](https://github.com/rust-lang/rust/files/13201439/config.toml.gz)

also I can provide you with the full build log: [build.log.gz](https://github.com/rust-lang/rust/files/13201449/build.log.gz)

**Comment by onur-ozkan** [comment]
> If I revert them both locally, I get a working x.py install

Did you wait for whole process(x.py install) to be finished? I don't understand how mkdir gets you into access denied error for `/usr/lib/rust/1.74.0_beta` path while you can actually create dirs/files in it..

**Comment by stefson** [comment]
Maybe there is some confusion about relative and absolute path? The access denied error is valid, as x.py never should mess with /usr/lib/ in install, but use /var/tmp/portage/dev-lang/rust-1.74.0_beta/image/ as it's prefix, if you want to name it like that.

**Comment by onur-ozkan** [comment]
True, there is a confusion between prefix and DESTDIR right now. I will fix it today.

**Comment by stefson** [comment]
Thank you, I'm happy to test any patch before you commit it to master.

**Comment by onur-ozkan** [comment]
> Thank you, I'm happy to test any patch before you commit it to master.

You can test #117383

**Comment by stefson** [comment]
yes this works for me, please also beta nominate

**Comment by onur-ozkan** [comment]
> @onur-ozkan has this been beta-nominated? I'm new to this process, it's just I can't see a rollup pullrequest anywhere and the release of 1.74.0 is coming closer every day.

It is https://github.com/rust-lang/rust/pull/117383#issuecomment-1786019764.

cc @rust-lang/release

**Comment by cuviper** [comment]
@onur-ozkan for backport, the PR's responsible team (T-bootstrap) also needs to label it beta-accepted. Once it has both labels, the release team will handle the actual backport.
